### ISSUE TITLE
Files cannot be deleted or updated

### ISSUE THREAD

**Issue by onestudy** [minio/minio#11592]
## Expected Behavior
The files can be deleted or changed through mc or minioclient.

## Current Behavior
Execute the following command and output the error message after waiting for a long time.
```bash
./mc rm minio1/report/1000000000/e5bd06b3f8234b37bf9c2f864fe21c39/6e9046bdcdbb465f823d08e4529821b9/b20484e81b1146f285afcac759a0efed/899b26b429a14781af3a5f67e77d2200/1000000000/jsformula.json --json
{
 "status": "success",
 "key": "minio1/report/1000000000/e5bd06b3f8234b37bf9c2f864fe21c39/6e9046bdcdbb465f823d08e4529821b9/b20484e81b1146f285afcac759a0efed/899b26b429a14781af3a5f67e77d2200/1000000000/jsformula.json",
 "versionID": "",
 "modTime": "0001-01-01T00:00:00Z",
 "size": 661594
}
{
 "status": "error",
 "error": {
  "message": "Failed to remove `minio1/report/1000000000/e5bd06b3f8234b37bf9c2f864fe21c39/6e9046bdcdbb465f823d08e4529821b9/b20484e81b1146f285afcac759a0efed/899b26b429a14781af3a5f67e77d2200/1000000000/jsformula.json`.",
  "cause": {
   "message": "A timeout occurred while trying to lock a resource, please reduce your request rate",
   "error": {
    "Code": "RequestTimeout",
    "Message": "A timeout occurred while trying to lock a resource, please reduce your request rate",
    "BucketName": "",
    "Key": "",
    "RequestID": "",
    "HostID": "",
    "Region": "",
    "Server": ""
   }
  },
  "type": "error",
  "sysinfo": {
   "host.arch": "amd64",
   "host.cpus": "8",
   "host.lang": "go1.15.7",
   "host.name": "CWCL-YY-T077099",
   "host.os": "linux",
   "mem.heap.total": "66 MB",
   "mem.heap.used": "2.2 MB",
   "mem.total": "75 MB",
   "mem.used": "2.2 MB"
  }
 }
}
```

## Context

A few files can't be deleted. Other files operate normally.
The distribution of the problem files on the disk is as follows:
1. Directory structure of host1:
|-- jsformula.json
|------xl.meta
|------e34943c8-ed83-4b62-9d1f-a4c416cc8fca
|----------------part.1

2. Directory structure of host2:
|-- jsformula.json
|------xl.meta
|------e34943c8-ed83-4b62-9d1f-a4c416cc8fca
|----------------part.1
|------fb059a57-1283-4b61-a20f-4290cb66fff1
|----------------part.1

3. Directory structure of host3:
|-- jsformula.json
|------xl.meta
|------e34943c8-ed83-4b62-9d1f-a4c416cc8fca
|----------------part.1

4. Directory structure of host4:
|-- jsformula.json
|------xl.meta
|------e34943c8-ed83-4b62-9d1f-a4c416cc8fca
|----------------part.1

**It seems that there is something wrong with host2. Is this the reason why it can't be deleted?**

## Your Environment

* Version used (`minio --version`):RELEASE.2021-02-14T04-01-33Z
* Server setup and configuration:
* Operating System and version (`uname -a`):Linux CWCL-YY-T077099 3.10.0-1062.el7.x86_64 #1 SMP Thu Jul 18 20:25:13 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
* Service start bash:
```bash
/data/minio/run/minio server --config-dir /data/minio/etc \
http://10.3.77.99/data/minio/data \
http://10.3.77.100/data/minio/data \
http://10.3.77.101/data/minio/data \
http://10.3.77.102/data/minio/data
```

**Comment by harshavardhana** [comment]
> ```shell
>     "Message": "A timeout occurred while trying to lock a resource, please reduce your request rate",
> ```

@onestudy this is because you have some locks held on the object please restart `mc admin service restart` and upload again.

**Comment by onestudy** [comment]
> > ```shell
> >     "Message": "A timeout occurred while trying to lock a resource, please reduce your request rate",
> > ```
> 
> @onestudy this is because you have some locks held on the object please restart `mc admin service restart` and upload again.

I don't have the operation of locking files. All files are uploaded through the upload method of minioclient. What is the reason for an extra 'fb059a57-1283-4b61-a20f-4290cb66fff1' directory and 'part. 1' file on the host2 server?
This problem often happens. I want to know how to avoid it instead of restarting the service?

**Comment by harshavardhana** [comment]
> I don't have the operation of locking files. All files are uploaded through the upload method of minioclient. What is the reason for an extra 'fb059a57-1283-4b61-a20f-4290cb66fff1' directory and 'part. 1' file on the host2 server?
> This problem often happens. I want to know how to avoid it instead of restarting the service?

This is because your client timesout and leaves locks on the server and MinIO is not able to clear it because client is perhaps still holding on to that lock. 

It basically also shows that your setup off has a poor network connection between servers and they often timeout. There will be messages regarding this in the MinIO server console output.

I don't think there is anything we can do to help here. We are improving the locking to have sort of self expiry in a new feature https://github.com/minio/minio/pull/11535

**Comment by harshavardhana** [comment]
Will be fixed by #11535
